### PR TITLE
Create IconButton and use it

### PR DESCRIPTION
**Pull request by kylemh**

### CODE REVIEW

File: packages/zephyr/src/Button.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import styled, { keyframes, useTheme } from 'styled-components';
 import { Box, BoxStylingProps } from './Box';
 import { ButtonVariantName } from './theme/variants/button';
 
-export type ButtonSize = 'large' | 'medium' | 'small';
+export type ButtonSize = 'xs' | 'small' | 'medium' | 'large';
```
Comment (andyhqtran) [inline packages/zephyr/src/Button.tsx:13]:
```suggestion
export type ButtonSize = 'extra-small' | 'small' | 'medium' | 'large';
```

rather have `x-small` or `extra-small` or have everything changed to `xs` `s` `m` `l`

File: packages/zephyr/src/IconButton.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        type={type}
+        variant={variant}
+      >
+        {/* If button is wrapped with tooltip, it doesn't require assistive text. It's already provided on focus via the tooltip. */}
+        {!hasTooltip && <VisuallyHidden>{hiddenLabel}</VisuallyHidden>}
+        <Box
+          as={icon}
+          tx={{
+            '& path': {
+              strokeWidth: 2,
```
Comment (andyhqtran) [inline packages/zephyr/src/IconButton.tsx:81]: We don't have this for all icons, maybe we should remove this and have them apply it on the `IconButton` component itself

**Review by andyhqtran — APPROVED** [review]

File: packages/zephyr/src/IconButton.tsx (hunk 2 above)
Comment (kylemh) [reply packages/zephyr/src/IconButton.tsx:81]: I guess I was going off of https://www.figma.com/file/EZ1hzsmsNYblYoBQYvVgEV/Web-Component-Library?node-id=2943%3A770

What would a good system be?

We could expose an `iconStyles` prop which is basically just a TXProp, but have the properties limited to things relevant to `path`s?

File: packages/zephyr/src/Button.tsx (hunk 1 above)
Comment (kylemh) [reply packages/zephyr/src/Button.tsx:13]: Will make the change!

File: packages/zephyr/src/IconButton.tsx (hunk 2 above)
Comment (kylemh) [reply packages/zephyr/src/IconButton.tsx:81]: Posted in design (for posterity)

**Review by tomaszczura — APPROVED** [review]

**Review by rnmp — APPROVED** [review]
Exciting

**Pull request merged**